### PR TITLE
feat(cheatcodes): `tryFfi` cheatcode

### LINKED ISSUE #2932: `forge test`/`forge script`: throw when ffi fails
### Component

Forge

### Describe the feature you would like

I would expect the script to halt and throw when `vm.ffi` fails. Currently it (almost) silently continues and prints "Script ran successfully."

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

Closes #2932 .

## Solution

Adds a `tryFfi` cheatcode, which returns a `FfiResult` struct which contains:
- The `exit_code`
- The `stdout` as bytes.
- The `stderr` as bytes.

### CODE REVIEW

File: testdata/cheats/TryFfi.t.sol [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    function testTryFfiFail() public {
+        string[] memory inputs = new string[](3);
+        inputs[0] = "bash";
+        inputs[1] = "-c";
+        inputs[2] =
+            "ls foo";
+
+        Cheats.FfiResult memory f = cheats.tryFfi(inputs);
```
Comment (Evalir) [inline testdata/cheats/TryFfi.t.sol:30]: Currently debugging why this is reverting with a cryptic `EVMError: Revert` error. While the `try_ffi` function for the cheatcode works well and returns the data normally, it seems to be causing a revert down the line.

**Review by mattsse — CHANGES_REQUESTED** [review]
sorry forgot to review this.

I like this, but it needs a rebase because we moved cheatcodes

**Review by mattsse — COMMENTED** [review]
looks like this just needs a rebase?

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
there's a failing test, otherwise lgtm

**Comment by Evalir** [comment]
yeah need to debug why this is failing—the cheatcode actually always reverts for some reason. need to test further before merge

**Pull request closed without merging**